programs/test/CMakeLists.txt

```diff
@@ -27,6 +27,15 @@ if(TEST_CPP)
     target_link_libraries(cpp_dummy_build ${mbedcrypto_target})
 endif()
 
+if(USE_SHARED_MBEDTLS_LIBRARY AND
+   NOT ${CMAKE_SYSTEM_NAME} MATCHES "[Ww][Ii][Nn]")
+    add_executable(dlopen "dlopen.c")
+    target_include_directories(dlopen PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+        target_link_libraries(dlopen "-ldl")
```

Comment:
```suggestion
        target_link_libraries(dlopen ${CMAKE_DL_LIBS})
```

Reply:
Thanks, I'd missed that.

It's not for Windows which doesn't have dlopen anyway. It's for non-Linux POSIX systems such as *BSD.